### ISSUE TITLE
[Bug] EAGLE accept_len is lower than vllm & SafeLab/EAGLE

### ISSUE THREAD

**Issue by xhdidi** [sgl-project/sglang#8228]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

We use the same draft model weight, drafting the same draft structure (--speculative-num-steps 2 --speculative-eagle-topk 1 --speculative-num-draft-tokens 2), the accept_length in vllm & [SafeAILab/EAGLE](https://github.com/SafeAILab/EAGLE) is greater than sglang, and the performance gain is also higher. Has anyone encountered similar problems?

### Reproduction

model: Qwen2-72B
command: python3 -m sglang.launch_server --model-path Qwen2-72B --tensor-parallel-size 8 --dtype bfloat16 --served-model-name Qwen2  --speculative-algorithm EAGLE --speculative-draft-model-path EAGLE-Qwen2-72B --speculative-num-steps 2 --speculative-eagle-topk 1 --speculative-num-draft-tokens 2 --mem-fraction 0.7

### Environment

Python: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
CUDA available: True
GPU 0,1,2,3,4,5,6,7: NVIDIA H20
GPU 0,1,2,3,4,5,6,7 Compute Capability: 9.0
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 12.2, V12.2.140
CUDA Driver Version: 535.161.07
PyTorch: 2.7.0+cu126
sglang: 0.4.9
sgl_kernel: 0.2.4
flashinfer_python: 0.2.7.post1
triton: 3.3.0
transformers: 4.53.0
torchao: 0.9.0
numpy: 1.26.4
aiohttp: 3.12.14
fastapi: 0.115.12
hf_transfer: 0.1.9
huggingface_hub: 0.33.4
interegular: 0.3.3
modelscope: 1.28.0
orjson: 3.9.14
outlines: 0.1.11
packaging: 23.2
psutil: 5.9.8
pydantic: 2.11.5
python-multipart: 0.0.20
pyzmq: 26.4.0
uvicorn: 0.27.1
uvloop: 0.19.0
vllm: 0.9.2
xgrammar: 0.1.19
openai: 1.82.0
tiktoken: 0.7.0
anthropic: 0.58.2
litellm: 1.74.7
decord: 0.6.0
NVIDIA Topology: 
        GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      NV18    NV18    NV18    NV18    NV18    NV18    NV18    0-47,96-143     0               N/A
GPU1    NV18     X      NV18    NV18    NV18    NV18    NV18    NV18    0-47,96-143     0               N/A
GPU2    NV18    NV18     X      NV18    NV18    NV18    NV18    NV18    0-47,96-143     0               N/A
GPU3    NV18    NV18    NV18     X      NV18    NV18    NV18    NV18    0-47,96-143     0               N/A
GPU4    NV18    NV18    NV18    NV18     X      NV18    NV18    NV18    48-95,144-191   1               N/A
GPU5    NV18    NV18    NV18    NV18    NV18     X      NV18    NV18    48-95,144-191   1               N/A
GPU6    NV18    NV18    NV18    NV18    NV18    NV18     X      NV18    48-95,144-191   1               N/A
GPU7    NV18    NV18    NV18    NV18    NV18    NV18    NV18     X      48-95,144-191   1               N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

ulimit soft: 1048576

**Comment by Swipe4057** [comment]
vllm reproduction info?

**Comment by xhdidi** [comment]
We found that sglang has a higher default sampling temperature, which causes the Logits to be different from vllm/safelab. After setting the sampling temperature to 0.0, the outputs of all hidden layers can be aligned, but the accepted_length is lower, almost 0.

**Comment by xhdidi** [comment]
> vllm reproduction info?

vllm serve Qwen2-72B --tensor-parallel-size 8 --use-v2-block-manager --enable-prefix-caching --dtype bfloat16 --gpu-memory-utilization 0.8 --speculative_config '{"model": "EAGLE-Qwen2-72B", "num_speculative_tokens": 2, "draft_tensor_parallel_size": 1}'
